### PR TITLE
faster bisect

### PR DESCRIPTION
**Pull request by thatbirdguythatuknownot**
it should also be more OS-independent (replacing `long` with `Py_ssize_t`)

### CODE REVIEW

File: submodule/main.c
Comment (juliusgeo) [inline submodule/main.c:10]: One super tiny nitpick is that you can use up to 100 chars for each line, which might make some of the comments cleaner.

Comment (thatbirdguythatuknownot) [reply submodule/main.c:10]: That specific comment is pretty nice and clean already but I changed some others.

**Review by juliusgeo — APPROVED** [review]
Nice! Thanks! I will update the benchmarks after you merge.

**Comment by thatbirdguythatuknownot** [comment]
> Nice! Thanks! I will update the benchmarks after you merge.

@juliusgeo I can't merge, I don't have write access.

**Pull request merged**
